### ISSUE TITLE
Duplicate SessionID

### ISSUE THREAD

**Issue by FattyMango** [quickfixgo/quickfix#614]
I'm creating a FIX(4.4) client and I'm trying to implement a retry mechanism that stops the entire initiator and starts it all over again, I'm doing this since the Quickfix auto reconnect doesn't send a `ResetOnLogon` in the reconnect message.

but when I try to create a new initiator with the same session I get a `Duplicate SessionID` error, I was able to reproduce the error and it appears to be from this function

```go
func registerSession(s *session) error {
	sessionsLock.Lock()
	defer sessionsLock.Unlock()

	if _, ok := sessions[s.sessionID]; ok { // here
		return errDuplicateSessionID
	}

	sessions[s.sessionID] = s
	return nil
}
```
it works as expected when I remove the session check.

**Comment by AlexandrosKyriakakis** [comment]
When you create a session ([here](https://github.com/quickfixgo/quickfix/blob/3bf2b871219bd9921c206adf59585b920c133538/initiator.go#L98-L98)) there is no way to remove it from this [sessions map](https://github.com/quickfixgo/quickfix/blob/3bf2b871219bd9921c206adf59585b920c133538/registry.go#L24-L24) so when you create another initiator with same configs under the same thread you got the error `Duplicate SessionID` 

We just need to remove the sessions map from global variables

**Comment by ackleymi** [comment]
Resolved with #637
